src/main/java/org/swisspush/redisques/RedisQues.java

```diff
@@ -297,28 +311,21 @@ public void run() {
             }
             void resume() {
                 try {
-                    long stepEpochMs = currentTimeMillis();
-                    int stepBeginIdx = i;
-                    while (iter.hasNext()) {
-                        var entry = iter.next();
-                        var queueName = entry.getKey();
-                        var dequeueStatistic = entry.getValue();
-                        if (entry.getValue().isMarkedForRemoval()) {
-                            RedisQues.this.dequeueStatistic.remove(queueName);
-                        }
-                        dequeueStatisticCollector.setDequeueStatistic(queueName, dequeueStatistic);
-                        i += 1;
-                        long nowEpochMs = currentTimeMillis();
-                        long stepDurationMs = nowEpochMs - stepEpochMs;
-                        if (stepDurationMs >= 8) {
-                            log.debug("Release EvLoop after step={}x, soFar={}x, size={}x statistics. Took step={}ms, total={}ms.",
-                                    i - stepBeginIdx, i, size, stepDurationMs, nowEpochMs - startEpochMs);
-                            vertx.runOnContext(v -> resume());
-                            return;
+                    List<Future<Void>> futureList = new ArrayList<>();
+                    vertx.executeBlocking((Handler<Promise<Void>>) promise -> {
```

Comment:
The relevant change is here, refer to https://vertx.io/docs/apidocs/io/vertx/core/Vertx.html#executeBlocking-io.vertx.core.Handler-boolean-io.vertx.core.Handler-